components/brave_new_tab_ui/hooks/PrefHookManager.ts

```diff
@@ -104,27 +104,25 @@ export class PrefHookManager<T> {
  * @returns A hook which can be used to subscribe to and update prefs managed by
  * |prefManager|.
  */
-export function createPrefsHook<OnType> (prefManager: PrefHookManager<OnType>) {
+export function createPrefsHook<OnType extends {}>(prefManager: PrefHookManager<OnType>) {
     return <PrefType extends keyof OnType>(pref: PrefType) => {
-        const [value, setValue] = useState(prefManager.getPref(pref))
+        const [, setValue] = useState(prefManager.getPref(pref))
         const setPref = useCallback((value: OnType[PrefType]) => prefManager.savePref(pref, value), [prefManager, pref])
         useEffect(() => {
             // Update the value here - the value could have been updated between
             // the hook being created and getting mounted (triggering the
             // useEffect).
             setValue(prefManager.getPref(pref))
 
-            const handler: PrefListener<OnType, PrefType> = (_, newValue) => {
-                setValue(newValue as typeof value)
-            }
+            const handler: PrefListener<OnType, PrefType> = (_, newValue) => setValue(newValue)
             prefManager.addPrefListener(pref, handler)
             return () => {
                 prefManager.removePrefListener(pref, handler)
             }
         }, [prefManager, pref])
 
         return [
-            value,
+            prefManager.getPref(pref),
```

Comment:
Q. I'm curios why this change was needed.

Reply:
There was a caching issue - not 100% sure what was causing it